### ISSUE TITLE
Cannot run inference - after conversion from keras model to tflite

### ISSUE THREAD

**Issue by Adman** [google-coral/edgetpu#218]
Hello,

I have a segmentation model taking input of size `(480, 640, 3)` with values in range from 0 to 1 and produces output of size `(480, 640, 1)`.
The conversion from keras model file to tflite went pretty fine. However, when I'm calling `run_inference` with flattened numpy array, it always raises `RuntimeError: external/org_tensorflow/tensorflow/lite/core/subgraph.cc:1044 required_bytes != bytes (3686400 != 921600`.
If I try to retype the input image to let's say uint32, other exception appears: `TypeError: Cannot cast array data from dtype('uint32') to dtype('uint8') according to the rule 'safe'`.
Does anyone experienced same issue and could help?

**Comment by Namburger** [comment]
Hello, that usually means that the model still has float input which our API doesn't support for the python interface yet.
Can you upload the model here?

**Comment by Adman** [comment]
[resnetsmall.zip](https://github.com/google-coral/edgetpu/files/5204735/resnetsmall.zip)
Here is the original keras model.

**Comment by Namburger** [comment]
Do you have the cpu tflite model?

**Comment by Adman** [comment]
Yes, here it is 
[resnetsmall.hdf5.zip](https://github.com/google-coral/edgetpu/files/5207784/resnetsmall.hdf5.zip)

**Comment by Adman** [comment]
What do you mean by "cpu"? Is there any option to convert the model to tflite cpu vs gpu?

**Comment by Namburger** [comment]
The tflite model after converting should be a CPU model before you compile it to an edgetpu model. Only edgetpu model can be run with our api

**Comment by Adman** [comment]
Is it really hdf5 ? Here is the code I use for converting the model
```python
def convert(path):
    custom_objects = { 
        'mean_iou': models.mean_iou
    }

    converter = tf.lite.TFLiteConverter.from_keras_model_file(
        path,
        custom_objects=custom_objects
    )
    tflite_model = converter.convert()

    with open('{}.tflite'.format(path), 'wb') as f:
        f.write(tflite_model)
```

**Comment by Namburger** [comment]
@Adman I see, it is in deed `.hdf5.tflite`, the hdf5 part confused me, sorry about that (I was still not in the office this morning when I saw the zip file, just wanted to respond earlier).
Any how, as I suspected, the model is not quantized and also not compiled for the edgetpu, please go through the full [post training quantization](https://www.tensorflow.org/lite/performance/post_training_quantization#integer_only) process instead of just converting the model. As of now it is still a float cpu model:
![218a](https://user-images.githubusercontent.com/25712149/92932914-2b825200-f40b-11ea-9f02-bb7c27da77b9.png)
Then after you have a fully quantized model, pass it through the [edgetpu compiler](https://coral.ai/docs/edgetpu/compiler/)

**Comment by Adman** [comment]
@Namburger Thanks for clarification. I managed to run the quantization based on that example, but I had to use newest tensorflow. However, when I tried to run edgetpu compiler, I got this message
```
Edge TPU Compiler version 14.1.317412892
Invalid model: trained_models/model.hdf5.tflite
Model not quantized
```

**Comment by Adman** [comment]
The code that I used for converting this time looks as follows:
```python
    custom_objects = {
        'mean_iou': models.mean_iou
    }
    model = tf.keras.models.load_model(path, custom_objects=custom_objects)


    converter = tf.lite.TFLiteConverter.from_keras_model(
        model,
        #custom_objects=custom_objects
    )
    converter.optimizations = [tf.lite.Optimize.DEFAULT]
    converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
    converter.inference_input_type = tf.int8  # or tf.uint8
    converter.inference_output_type = tf.int8  # or tf.uint8

    def representative_dataset_gen():
        num_calibration_steps = 50

        data = train_generator(1,
                               train_dir,
                               'image',
                               'masks',
                               img_target_size=IMG_TARGET_SIZE,
                               aug=False)

        inp = []
        for i, j in enumerate(data):
            if i == num_calibration_steps:
                break
            inp.append(j[0].astype(np.float32))

        for i in range(num_calibration_steps):
            # Get sample input data as a numpy array in a method of your choosing.
            yield [inp[i]]

    converter.representative_dataset = representative_dataset_gen
    tflite_model = converter.convert()

    with open('{}.tflite'.format(path), 'wb') as f:
        f.write(tflite_model)
```

**Comment by Namburger** [comment]
Hello, what tensorflow version are you using? 
For tf >= 2.3*, the input size isn't static, we'll start documenting this soon, but can you add this line before instantiating the converter?

```
model.input.set_shape((1,) + model.input.shape[1:]) # this line

converter = tf.lite.TFLiteConverter.from_keras_model(
        model,
    )
```

**Comment by Adman** [comment]
Hello, thanks for helpful advices. It started working, but compiling the tflite model raises that it does not recognize `RESIZE_NEAREST_NEIGHBOR` operation. I found this issue https://github.com/google-coral/edgetpu/issues/152 and based on that I downgraded tf version to 2.1, which solved the problem. The only thing which is problematic is prediction, which always returns zeros for all the pixels in the picture (semantic segmentation).
